### PR TITLE
Fixed check for volume source being null or empty

### PR DESCRIPTION
**Pull request by piccobit**
This bug was hidden by the second condition which is normally always false and was only discovered when I disabled the second condition to allow the usage of named volumes which don't start with a '/'.

### CODE REVIEW

File: helios-client/src/main/java/com/spotify/helios/common/JobValidator.java [hunk 1]
```diff
@@ -150,7 +150,7 @@ public JobValidator(final boolean shouldValidateJobHash,
         errors.add("Volume path is not absolute: " + path);
         continue;
       }
-      if (!isNullOrEmpty(source) && !source.startsWith("/")) {
+      if (isNullOrEmpty(source) && !source.startsWith("/")) {
```
Comment (rohansingh) [inline helios-client/src/main/java/com/spotify/helios/common/JobValidator.java]: Are you sure this logic wasn't correct before? Wouldn't this now throw an NPE if `source` is null?

Comment (piccobit) [inline helios-client/src/main/java/com/spotify/helios/common/JobValidator.java]: @rohansingh Ah, yes, you're right! In our use case with the named volume I've removed the second condition completely and there it makes sense!

Comment (piccobit) [inline helios-client/src/main/java/com/spotify/helios/common/JobValidator.java]: Ok, I've separated the checks for the volume source to prevent the NPE.

File: helios-client/src/main/java/com/spotify/helios/common/JobValidator.java [hunk 2]
```diff
@@ -150,7 +150,11 @@ public JobValidator(final boolean shouldValidateJobHash,
         errors.add("Volume path is not absolute: " + path);
         continue;
       }
-      if (!isNullOrEmpty(source) && !source.startsWith("/")) {
```
Comment (davidxia) [inline helios-client/src/main/java/com/spotify/helios/common/JobValidator.java]: We can just change `&&` to `||`, but I guess more specific errors can't hurt.

File: helios-client/src/main/java/com/spotify/helios/common/JobValidator.java [hunk 3]
```diff
@@ -150,7 +150,11 @@ public JobValidator(final boolean shouldValidateJobHash,
...
+      if (isNullOrEmpty(source)) {
```
Comment (mattnworb) [inline helios-client/src/main/java/com/spotify/helios/common/JobValidator.java]: it seems like before, a null or empty source was allowed. Now this will cause an error. Am I correct on that and is that change intentional?

**Comment by mattnworb** [comment]
we seem to mention in https://github.com/spotify/helios/blob/master/docs/user_manual.md#volumes that the right-hand side of the volume can be left off or empty

**Comment by davidxia** [comment]
Ah, the original code makes sense then.

File: helios-client/src/main/java/com/spotify/helios/common/JobValidator.java (hunk 3 above)
Comment (mattnworb) [inline helios-client/src/main/java/com/spotify/helios/common/JobValidator.java]: @piccobit thanks - it would be helpful to see both changes in context.
